### PR TITLE
fix: update config parsing structure

### PR DESCRIPTION
**Pull request by omerhabib26**
### Description

- Update dictionary for ProgramConfig
- Update UI related feature flags under a single Key
- fix image load in course

fix: Issue-[318](https://github.com/openedx/openedx-app-android/issues/318)

### CODE REVIEW

**Review by HamzaIsrar12 — DISMISSED** [review]
LGTM 🏎️ 

_Please request a feedback from Volodymyr as well._

**Review by k1rill — APPROVED** [review]

**Comment by omerhabib26** [comment]
@volodymyr-chekyrta please spare some time to review this PR.
Thanks.

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM

**Comment by volodymyr-chekyrta** [comment]
Any chance to merge it after the https://github.com/openedx/openedx-app-android/pull/308 changes?

**Review by dixidroid — APPROVED** [review]

**Comment by omerhabib26** [comment]
@volodymyr-chekyrta sure we can merge it after [PR-308](https://github.com/openedx/openedx-app-android/pull/308) and update the config structure accordingly.

**Comment by volodymyr-chekyrta** [comment]
> @volodymyr-chekyrta sure we can merge it after [PR-308](https://github.com/openedx/openedx-app-android/pull/308) and update the config structure accordingly.

Thank you!

File: core/src/main/java/org/openedx/core/config/Config.kt [hunk 1]
```diff
@@ -165,8 +157,7 @@ class Config(context: Context) {
         private const val DISCOVERY = "DISCOVERY"
         private const val PROGRAM = "PROGRAM"
         private const val BRANCH = "BRANCH"
-        private const val COURSE_NESTED_LIST_ENABLED = "COURSE_NESTED_LIST_ENABLED"
-        private const val COURSE_UNIT_PROGRESS_ENABLED = "COURSE_UNIT_PROGRESS_ENABLED"
+        private const val UI_COMPONENT = "UI_COMPONENT"
```
Comment (HamzaIsrar12) [inline core/src/main/java/org/openedx/core/config/Config.kt:160]: Please rename it to `UI_COMPONENTS`.

**Review by HamzaIsrar12 — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**